controllers/applicationvolumereplication_controller.go

```diff
@@ -120,106 +126,263 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 
+	r.Log.Info("Process subscriptions", "total", len(subscriptionList.Items))
+
 	requeue := false
 
-	for _, subscription := range subscriptionList.Items {
+	for idx, subscription := range subscriptionList.Items {
 		// On the hub ignore any managed cluster subscriptions, as the hub maybe a managed cluster itself.
 		// SubscriptionSubscribed means this subscription is child sitting in managed cluster
 		// Placement.Local is true for a local subscription, and can be used in the absence of Status
 		if subscription.Status.Phase == subv1.SubscriptionSubscribed ||
 			(subscription.Spec.Placement != nil && subscription.Spec.Placement.Local != nil &&
 				*subscription.Spec.Placement.Local) {
-			r.Log.Info("skipping local subscription", "name", subscription.Name)
+			r.Log.Info("Skipping local subscription", "name", subscription.Name)
 
 			continue
 		}
 
-		// Create a ManifestWork that represents the VolumeReplicationGroup CR for the hub to
-		// deploy on the managed cluster. A manifest workload is defined as a set of Kubernetes
-		// resources. The ManifestWork must be created in the cluster namespace on the hub, so that
-		// agent on the corresponding managed cluster can access this resource and deploy on the
-		// managed cluster. We create one ManifestWork for each VRG CR.
-		homeCluster, peerCluster, err := r.createVRGManifestWork(subscription)
-		if err != nil {
-			r.Log.Info("failed to create or update ManifestWork for", "subscription", subscription.Name, "error", err)
+		placementDecision, needRequeue := r.processSubscription(avr, &subscriptionList.Items[idx])
 
+		if needRequeue {
 			requeue = true
 
 			continue
 		}
 
-		placementDecisions[subscription.Name] = &ramendrv1alpha1.SubscriptionPlacementDecision{
-			HomeCluster: homeCluster,
-			PeerCluster: peerCluster,
+		if placementDecision != nil {
+			placementDecisions[subscription.Name] = placementDecision
 		}
-
-		r.Log.Info("created VolumeReplicationGroup manifest for ", "subscription", subscription.Name,
-			"HomeCluster", homeCluster, "PeerCluster", peerCluster)
 	}
 
+	r.Log.Info("Returning Placement Decisions", "Total", len(placementDecisions))
+
 	return placementDecisions, requeue
 }
 
-func (r *ApplicationVolumeReplicationReconciler) updateAVRStatus(
-	ctx context.Context,
+func (r *ApplicationVolumeReplicationReconciler) processSubscription(
 	avr *ramendrv1alpha1.ApplicationVolumeReplication,
-	placementDecisions ramendrv1alpha1.SubscriptionPlacementDecisionMap) error {
-	r.Log.Info("updating AVR status")
+	subscription *subv1.Subscription) (*ramendrv1alpha1.SubscriptionPlacementDecision, bool) {
+	const requeue = true
+	// Check to see if this subscription is paused for DR. If it is, then restore PVs to the new destination
+	// cluster, unpause the subscription, and skip it until the next reconciler iteration
+	if r.isSubsriptionPausedForDR(subscription.GetLabels()) {
+		if err := r.processPausedSubscription(avr, subscription); err != nil {
```

Comment:
(question) A prior paused subscription once unpaused will have the home/peer clusters in the status flipped?

It seems that way, as first we unpause, the next reconcile loop would go through the unpaused workflow, where home would be where the subscription is now placed.

Not stating this is an issue, just thinking what it may mean from a reasoning standpoint.

Reply:
That's right.  As an example:
```
// Initial Deployment
PlacementRule {A, B}. PlacementRule.Status.Decision == A
Subscription.Status.Statuses[homeCluster] == A

// Failover occurs
PlacementRule {A, B}. PlacementRule.Status.Decision == B
// Subscription is Paused
Subscription.Status.Statuses[homeCluster] == A
// RamenHub is invoked because of user action (changed CR for now)
AVR.FailoverClusters[Subscription] == B
RamenHub->restorePVs to B
RamendHub->UNPAUSE subscription and requeue for another reconciler call
Subscription gets unpaused and Subscription.Status.Statuses[homeCluster] == B
RamenHub->processes the unpause subscription just like the initial deploy
```